include/ginkgo/core/base/array.hpp

```diff
@@ -90,14 +101,25 @@ class Array {
 
     /**
      * The default deleter type used by Array.
+     * For const values, we need to use a deleter that takes a const array.
+     * This does not make much sense (read-only uninitialized memory), but helps
+     * keep the interface simple.
```

Comment:
If we assume that an const array can't own its data, this would not be necessary.
```suggestion
```